app/enhancements/enhancement.rb

```diff
@@ -0,0 +1,17 @@
+##
+# Enhancements take a SolrProperty and return another SolrProperty with new
+# values.
+class Enhancement
+  attr_reader :raw_property
+
+  # @param [SolrProperty] raw_property The property to transform
```

Comment:
Ah, we need interfaces in Ruby :)

I say ditch it if you don't actually need it.  I love the explicit explanation, but I don't like dead code we have to maintain and document.  Maybe instead, explain the interface in the docs for CompositeEnhancementsFactory?